jaeger_client/metrics.py

```diff
@@ -21,69 +21,122 @@
+        """
+        return self._noop
 
-    TRACES_JOINED_NOT_SAMPLED = '%s.traces-joined.not-sampled' % prefix
-    """Number of externally started non-sampled traces this tracer joined"""
+    def create_gauge(self, name, tags=None):
+        """
+        Generates a new gauge from the given name and tags and returns
+        a callable function used to update the gauge.
+        :param name: name of the gauge
+        :param tags: tags for the gauge
+        :return: a callable function which takes the value to update
+        the gauge with ie. def update(value)
+        """
+        return self._noop
 
-    SPANS_SAMPLED = '%s.spans.sampled' % prefix
-    """Number of sampled spans started/finished by this tracer"""
+    def _merge_tags(self, tags=None):
+        if not self._tags:
+            return tags
+        tags_cpy = self._tags.copy()
+        tags_cpy.update(tags)
+        return tags_cpy
 
-    SPANS_NOT_SAMPLED = '%s.spans.not-sampled' % prefix
-    """Number of not-sampled spans started/finished by this tracer"""
 
-    TRACER_DECODING_ERRORS = '%s.decoding.errors' % prefix
-    """Number of errors decoding tracing context"""
+class LegacyMetricsFactory(MetricsFactory):
+    """A MetricsFactory wrapper around Metrics."""
 
-    REPORTER_SUCCESS = '%s.spans.reported' % prefix
-    """Number of spans successfully reported"""
+    def __init__(self, metrics, tags=None):
+        super(LegacyMetricsFactory, self).__init__(
+            tags=tags
+        )
+        self._metrics = metrics
 
-    REPORTER_FAILURE = '%s.spans.failed' % prefix
-    """Number of spans in failed attempts to report"""
+    def create_counter(self, name, tags=None):
+        key = self._get_key(name, self._merge_tags(tags))
 
-    REPORTER_DROPPED = '%s.spans.dropped' % prefix
-    """Number of spans dropped from the reporter"""
+        def increment(value):
+            return self._metrics.count(key, value)
+        return increment
 
-    REPORTER_SOCKET = '%s.spans.socket_error' % prefix
-    """Number of spans dropped due to socket error"""
+    def create_timer(self, name, tags=None):
+        key = self._get_key(name, self._merge_tags(tags))
 
-    REPORTER_QUEUE_LENGTH = '%s.queue_length' % prefix
-    """Current number of spans in the reporter queue"""
+        def record(value):
+            return self._metrics.timing(key, value)
+        return record
 
-    SAMPLER_ERRORS = '%s.sampler.errors' % prefix
-    """Number of times the Sampler failed to retrieve sampling strategy"""
+    def create_gauge(self, name, tags=None):
+        key = self._get_key(name, self._merge_tags(tags))
 
-    def __init__(self, count=None, gauge=None):
+        def update(value):
+            return self._metrics.gauge(key, value)
+        return update
+
+    def _get_key(self, name, tags=None):
+        if not tags:
+            return name
+        key = name
+        for k in sorted(tags.iterkeys()):
+            key = key + '|' + str(k) + '=' + str(tags[k])
```

Comment:
not sure this is how we want to make the key anymore. In the expvar bridge I had to do a similar toString for the tags, and the format I used was `{name}.{tag-key}_{tag-value}..`, with tags sorted by keys. This makes it relatively friendly to grafana path-based functions, and avoids `|` and `=` which might be reserved characters in some metrics systems.